### ISSUE TITLE
Custom metrics: use the timestamp from sample

### ISSUE THREAD

**Issue by olivierlemasle** [kubernetes-sigs/prometheus-adapter#545]
For custom metrics, the timestamp is `time.Now()`; a `//TODO` was added 4 years ago to use the right timestamp instead:

https://github.com/kubernetes-sigs/prometheus-adapter/blob/89584579687e4a66f352a1ba26e957951de628e7/pkg/custom-provider/provider.go#L101-L102

We should use the timestamp from the sample, as it is done for external metrics and resource metrics.

**Comment by impact-maker** [comment]
@olivierlemasle Hi, I have been working on this issue and submited a pull request #554 for the same. I am a newly introduced developer for this organisation and would need some inputs to contribute for the same. Kindly let me know if there is any guide to read before the contributions or any rules to look for testing the code for the environment.

**Comment by olivierlemasle** [comment]
Thank you and welcome @impact-maker 

First, you'll find the Kubernetes Contributing Guidelines in [CONTRIBUTING.md](https://github.com/kubernetes-sigs/prometheus-adapter/blob/master/CONTRIBUTING.md). It also includes a guide on how to sign the CLA, which is required for any contribution, and general guidelines (how to write a good commit message, etc.). https://github.com/kubernetes/community/blob/master/contributors/guide/pull-requests.md is also a useful general guide.

From a technical perspective, for prometheus-adapter, you can use `make verify test` to run the linters / checkers and execute the tests locally, before pushing the code to GitHub.
You can use `make container` to build a container image of prometheus-adapter and test if everything is ok.

I'll review your PR later today.

**Comment by impact-maker** [comment]
@olivierlemasle  Thank you for guiding me through this process. I have tried to do a fair bit of research to go through the documentations provided. 
I have made another commit in the last pull request. Kindly review that and help me cut through the steep learning curve of contributing to this project. 
I have tried my best to run make verify test command but failed to do so and I am still not able to compile/ test the code before commiting. Your inputs on the same would really help me provide useful code contributions for this issue.
